cmd/dlt/oidcprovider/cmd.go

```diff
@@ -78,16 +78,19 @@ func run(cmd *cobra.Command, argv []string) {
 		os.Exit(1)
 	}
 
-	clusterID := clusterKey
 	if sub != nil {
-		clusterID = sub.ClusterID()
+		clusterKey = sub.ClusterID()
 	}
-	c, err := r.OCMClient.GetClusterByID(clusterID, r.Creator)
+	c, err := r.OCMClient.GetCluster(clusterKey, r.Creator)
```

Comment:
I was looking into this and it seems that this function already looks up the subscription as well. I'm thinking we can simplify this whole flow doing only GetCluster instead of also doing lines from 70 through 79 too. What do you think?

Reply:
This was actually my first thought as well.  I even opened a PR for it, and then closed it after I discovered why the sub lookup is important.

`GetCluster` will not return a cluster that has been deprovisioned, so you cannot use it to lookup the ClusterId of a cluster that no longer exists.  Only the subscription can tell you what the ID was for a deprovisioned cluster.

We can't rely on just the subscription returned from `GetClusterUsingSubscription` either, because this function will only return subscriptions where `status = 'Deprovisioned'`, it will not return active clusters.

I thought about changing the logic of `GetClusterUsingSubscription` to not filter by `status = 'Deprovisioned'`, but since this is also used by the `rosa delete operator-roles` command, and that command is behaving properly, I thought it better to provide the logic fixes to `rosa delete oidc-provider` instead.

Not filtering by `status = 'Deprovisioned'` would also change current behavior of the `rosa delete oidc-provider` command.  With the current behavior, if you have 1 active cluster, and 1 deprovisioned cluster, both using the same name, the `rosa delete oidc-provider` command will find and delete the correct OIDC provider.   If `GetClusterUsingSubscription` were modified to no longer filter by `status = 'Deprovisioned'`, then an active and deprovisioned cluster existing with the same name would now result in `errors.Conflict` being thrown.

This would cause the command to fail in a scenario, where it was previously successful.